### PR TITLE
Adds translation of http2.md to Portuguese

### PR DESCRIPTION
**Pull request by elaynelemos**
Makes progress on #505

Hey, guys! Sorry for being kinda late. It's a large article. hahah

That's it. Reach me out if you have any doubts (first contribution in HTTP Archive). :smile_cat:

### CODE REVIEW

**Comment by bazzadp** [comment]
Yeah that author does like to go on a bit 😁

File: src/content/pt/2019/http2.md [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQLxLA5Nojw28P7ceisqti3oTmNSM-HIRIR0bDb2icJS5TzONvRhdqxQcooh_45TmK97XVpot4kEQA0/pubchart?oid=466353517&format=interactive"
+  )
+}}
+
+A Figura 20.13 nos mostra quais recursos são mais comumente enviados. JavaScript e CSS são a esmagadora maioria dos itens enviados, tanto por volume quanto por bytes. Depois disso, há uma variedade de imagens, fontes e dados. No final, vemos cerca de 100 sites enviando vídeo, o que pode ser intencional ou pode ser um sinal de excesso de envio dos tipos de recurso errados!
+
+Uma preocupação levantada por alguns é que as implementações HTTP/2 redesignaram o cabeçalho `preload` HTTP` link` como um sinal para o push. Um dos usos mais populares da dica de recurso `preload` é informar o navegador sobre recursos descobertos posteriormente, como fontes e imagens, que o navegador não perceberá até que o CSS tenha sido solicitado, baixado e analisado. Se agora eles são enviados com base nesse cabeçalho, havia uma preocupação de que reutilizá-los poderia resultar em muitos envios indesejados.
```
Comment (bazzadp) [inline src/content/pt/2019/http2.md:279]: Fixing linting error:

```suggestion
Uma preocupação levantada por alguns é que as implementações HTTP/2 redesignaram o cabeçalho `preload` HTTP `link` como um sinal para o push. Um dos usos mais populares da dica de recurso `preload` é informar o navegador sobre recursos descobertos posteriormente, como fontes e imagens, que o navegador não perceberá até que o CSS tenha sido solicitado, baixado e analisado. Se agora eles são enviados com base nesse cabeçalho, havia uma preocupação de que reutilizá-los poderia resultar em muitos envios indesejados.
```

File: src/content/pt/2019/http2.md [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+
+[As propostas de informar o servidor sobre o status do cache do navegador foram bloqueadas](https://lists.w3.org/Archives/Public/ietf-http-wg/2019JanMar/0033.html) especialmente em questões de privacidade. Mesmo sem esse problema, existem outros problemas potenciais se o push não for usado corretamente. Por exemplo, enviar imagens grandes e, portanto, impedir o envio de CSS e JavaScript essenciais levará a sites mais lentos do que se você nem tivesse feito push!
+
+Também há muito pouca evidência até o momento de que o push, mesmo quando implementado corretamente, resulta no aumento de desempenho prometido. Esta é uma área em que, novamente, o HTTP Archive não está na melhor posição para responder, devido à natureza de como ele é executado (um rastreamento de sites populares usando o Chrome em um estado), então não vamos nos aprofundar muito aqui. No entanto, basta dizer que os ganhos de performance estão longe de ser claros e os problemas potenciais são reais.
+
+Putting that aside let's look at the usage of HTTP/2 push.
```
Comment (bazzadp) [inline src/content/pt/2019/http2.md:245]: Missed a translation here

File: src/content/pt/2019/http2.md [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+
+[Patrick Meenan](https://twitter.com/patmeenan) created [an example test page](https://github.com/pmeenan/http2priorities/tree/master/stand-alone), which deliberately tries to download a load of low priority, off-screen images, before requesting some high priority on-screen images. A good HTTP/2 server should be able to recognize this and send the high priority images shortly after requested, at the expense of the lower priority images. A poor HTTP/2 server will just respond in the request order and ignore any priority signals. [Andy Davies](./contributors#andydavies) has [a page tracking the status of various CDNs for Patrick's test](https://github.com/andydavies/http2-prioritization-issues). The HTTP Archive identifies when a CDN is used as part of its crawl, and merging these two datasets can tell us the percent of pages using a passing or failing CDN.
```
Comment (bazzadp) [inline src/content/pt/2019/http2.md:300]: Missed translating this paragraph

**Review by bazzadp — CHANGES_REQUESTED** [review]
Looks pretty good @elaynelemos -  did get that right this time didn't I? 😉

Spotted a few minor things to fix.

Could you also do two more things for me please:

First up, a quote from this text is in the `featured_chapters.html` file. Could you check the translation is roughly the same and if not update one or the other to bring them in line?

https://github.com/HTTPArchive/almanac.httparchive.org/blob/875346fe8c8d8dbffc7d723181b1cd38a165ab47/src/templates/pt/2019/featured_chapters.html#L65-L66

And secondly can you add yourself to the src/config/2019.json so you get credit for this translation on our Contributors page!

Thanks,
Barry

File: src/content/pt/2019/http2.md (hunk 2 above)
Comment (elaynelemos) [reply src/content/pt/2019/http2.md]: Wow! hahahah
I read it many times and even then missed out those translations. It's fixed now. Thanks!

File: src/content/pt/2019/http2.md (hunk 3 above)
Comment (elaynelemos) [reply src/content/pt/2019/http2.md]: Done. :D

**Review by bazzadp — CHANGES_REQUESTED** [review]
From a Technical perspective the translation looks good now! Ideally someone who speaks Portuguese can review the language itself.

However still need your details in `src/config/2019.json`. This file is alphabetical by first name so add your profile in at line 473.

Also can you "turn on" this chapter's featured quote in the home page by changing the first line of `src/templates/pt/featured_chapters.html` from this:

```
{%- set featured_chapter = ("javascript","javascript") | random %}
```

to this:
```
{%- set featured_chapter = ("javascript","http2") | random %}
```

**Review by bazzadp — CHANGES_REQUESTED** [review]
Just one small change required, then I can approve!

**Review by bazzadp — APPROVED** [review]
Looks good to me technically.

@HTTPArchive/portuguese-translators can one of you check if the translation is all OK?

File: src/content/pt/2019/http2.md [hunk 4]
```diff
@@ -0,0 +1,367 @@
+---
+part_number: IV
+chapter_number: 20
+title: HTTP/2
+description: Capítulo de HTTP/2 de 2019 do Web Almanac que cobre a adoção e o impacto do HTTP/2, HTTP/2 Push, Problemas do HTTP/2 e HTTP/3.
+authors: [bazzadp]
+reviewers: [bagder, rmarx, dotjs]
+translators: [elaynelemos]
```
Comment (bazzadp) [inline src/content/pt/2019/http2.md:8]: Add Analysts added in #1334 

```suggestion
translators: [elaynelemos]
analysts: [paulcalvano]
```

File: src/templates/pt/2019/featured_chapters.html [hunk 5]
```diff
@@ -1,4 +1,4 @@
-{%- set featured_chapter = ("javascript","javascript") | random %}
+{%- set featured_chapter = ("javascript","http2") | random %}
```
Comment (bazzadp) [inline src/templates/pt/2019/featured_chapters.html:1]:
```suggestion
{%- set featured_chapter = ("javascript","accessibility","http2") | random %}
```

**Comment by elaynelemos** [comment]
@bazzadp could we label this PR as hacktoberfest-accepted?

**Comment by elaynelemos** [comment]
> > @bazzadp could we label this PR as hacktoberfest-accepted?
> 
> Done. Hopefully this won't lead to a lot of spam!

I believe it will be ok. Because it's in the PR scope. But you can revoke if you think it would lead to any problems. :innocent:

**Comment by bazzadp** [comment]
Yeah hopefully should be fine. We are definitely open to new joiners so if that encourages quality contributions then I'm all for it. Just been reading the bad press lately! 😀

@HakaCode / @eduqg will either of you get a chance to review this translation? I'm itching to merge and release this (as well as [the Accessibility translation](https://github.com/HTTPArchive/almanac.httparchive.org/pull/1336)) but would be great to get a native speaker to review the language first if we can.

File: src/content/pt/2019/http2.md [hunk 6]
```diff
@@ -0,0 +1,368 @@
...
+queries: 20_HTTP_2
+published: 2019-11-11T00:00:00.000Z
+last_updated: 2020-09-21T00:00:00.000Z
+---
+
+## Introdução
+O HTTP/2 foi a primeira grande atualização do principal protocolo de transporte da web em quase 20 anos. Ele chegou com muitas expectativas: prometia um aumento de desempenho gratuito e sem desvantagens. Mais do que isso, poderíamos deixar de lado todos as adaptações e saídas mirabolantes a que o HTTP/1.1 nos forçava, devido às suas ineficiências. Técnicas como bundling, spriting, inlining e até mesmo sharding se tornariam não canônicas em um mundo com HTTP/2, visto que a performance otimizada já seria fornecida por padrão.
+
+Isso significava que mesmo aqueles sem as habilidades e os recursos para se concentrar na [performance na web](./performance) de repente teriam sites com bom desempenho. Contudo, a realidade foi, como sempre, um pouco mais matizada que isso. Já se passaram mais de quatro anos desde a aprovação formal do HTTP/2 como um padrão em maio de 2015 como [RFC 7540](https://tools.ietf.org/html/rfc7540), assim agora é um bom momento para verificar como essa tecnologia relativamente nova se saiu no mundo real.
```
Comment (eduqg) [inline src/content/pt/2019/http2.md:20]:
```suggestion
Isso significava que mesmo aqueles sem as habilidades e os recursos para se concentrar na [performance na web](./performance) de repente teriam sites com bom desempenho. Contudo, a realidade foi, como sempre, um pouco mais sutil do que isso. Já se passaram mais de quatro anos desde a aprovação formal do HTTP/2 como um padrão em maio de 2015 como [RFC 7540](https://tools.ietf.org/html/rfc7540), assim agora é um bom momento para verificar como essa tecnologia relativamente nova se saiu no mundo real.
```

Just a suggestion to review the nuanced word, I think maybe people will not understand it if the translation is 'matizado'. 

In google, define: nuance
2.
POR ANALOGIA
diferença sutil entre coisas, mais ou menos similares, postas em contraste; matiz, sutileza.

File: src/content/pt/2019/http2.md [hunk 7]
```diff
@@ -0,0 +1,368 @@
...
+  image="ch20_fig2_http2_usage_by_request.png",
+  alt="Uso do HTTP/2 por requisição.",
+  caption='Uso de HTTP/2 por requisição. (Fonte: <a href="https://httparchive.org/reports/state-of-the-web#h2">HTTP Archive</a>)',
+  description="Gráfico da série temporal mostrando a adoção de HTTP/2 em 55% para desktops e dispositivos móveis em julho de 2019. A tendência está crescendo continuamente em cerca de 15 pontos por ano.",
+  width=600,
+  height=321
+  )
+}}
+
+The results show that HTTP/2 usage is now the majority protocol-an impressive feat just 4 short years after formal standardization! Looking at the breakdown of all HTTP versions by request we see the following:
```
Comment (eduqg) [inline src/content/pt/2019/http2.md:82]: Maybe you forgot to translate :smile:

File: src/content/pt/2019/http2.md [hunk 8]
```diff
@@ -0,0 +1,368 @@
...
+
+Tudo isso antes de entrarmos nos poucos sites que recomendam a atualização para `http1.0`, `http://1.1`, ou mesmo `-all,+TLSv1.3,+TLSv1.2`. Existem claramente alguns erros de digitação nas configurações do servidor web acontecendo aqui!
+
+Existem outros problemas além de implementação que podemos analisar. Por exemplo, HTTP/2 é muito mais restrito sobre os nomes de cabeçalho HTTP, rejeitando a requisição inteira se você responder com espaços, dois pontos ou outros nomes de cabeçalho HTTP inválidos. Os nomes de cabeçalho também são convertidos em letras minúsculas, o que pega alguns de surpresa se sua aplicação assumir certa capitalização. Isso nunca foi garantido anteriormente, já que o HTTP/1.1 afirma especificamente que [os nomes dos cabeçalhos não diferenciam maiúsculas de minúsculas](https://tools.ietf.org/html/rfc7230#section-3.2), mas ainda assim alguns dependeram disso. O HTTP Archive também pode ser usado para identificar esses problemas, embtivoora alguns deles não sejam aparentes na página inicial, mas não nos aprofundamos nisso este ano.
```
Comment (eduqg) [inline src/content/pt/2019/http2.md:339]:
```suggestion
Existem outros problemas além de implementação que podemos analisar. Por exemplo, HTTP/2 é muito mais restrito sobre os nomes de cabeçalho HTTP, rejeitando a requisição inteira se você responder com espaços, dois pontos ou outros nomes de cabeçalho HTTP inválidos. Os nomes de cabeçalho também são convertidos em letras minúsculas, o que pega alguns de surpresa se sua aplicação assumir certa capitalização. Isso nunca foi garantido anteriormente, já que o HTTP/1.1 afirma especificamente que [os nomes dos cabeçalhos não diferenciam maiúsculas de minúsculas](https://tools.ietf.org/html/rfc7230#section-3.2), mas ainda assim alguns dependeram disso. O HTTP Archive também pode ser usado para identificar esses problemas, embora alguns deles não sejam aparentes na página inicial, mas não nos aprofundamos nisso este ano.
```

File: src/content/pt/2019/http2.md [hunk 9]
```diff
@@ -0,0 +1,368 @@
...
..., todos têm um papel a desempenhar para tornar isso mais fácil. Atrelar software a sistemas operacionais sempre aumenta o tempo de implantação. De fato, um dos principais motivos a favor o QUIC é quebrar uma barreira similar com a implantação de alterações do TCP. Em muitos casos, não há razão real para vincular as versões do servidor web aos sistemas operacionais. O Apache (para usar um dos exemplos mais populares) rodará com suporte a HTTP/2 em sistemas operacionais mais antigos, mas obter uma versão atualizada no servidor não deve exigir a experiência ou o risco que exige atualmente. O Nginx se sai muito bem aqui, hospedando repositórios para as distribuições comuns de Linux para tornar a instalação mais fácil, e se a equipe do 6Apache (ou os fornecedores de distribuição Linux) não oferecerem algo semelhante, então eu só posso ver o uso do Apache continuando a diminuir enquanto luta para manter a relevância e mudar sua reputação de ser antigo e lento (com base em instalações mais antigas), embora as versões atualizadas tenham uma das melhores implementações do HTTP/2. Eu vejo isso como um problema menor para o IIS, já que geralmente é o servidor web preferido no lado do Windows.
```
Comment (eduqg) [inline src/content/pt/2019/http2.md:362]:
```suggestion
Também falamos sobre como (ainda!) não é fácil conseguir suporte ao HTTP/2 em algumas instalações. Desenvolvedores de servidores, distribuidores de sistemas operacionais e clientes finais, todos têm um papel a desempenhar para tornar isso mais fácil. Atrelar software a sistemas operacionais sempre aumenta o tempo de implantação. De fato, um dos principais motivos a favor o QUIC é quebrar uma barreira similar com a implantação de alterações do TCP. Em muitos casos, não há razão real para vincular as versões do servidor web aos sistemas operacionais. O Apache (para usar um dos exemplos mais populares) rodará com suporte a HTTP/2 em sistemas operacionais mais antigos, mas obter uma versão atualizada no servidor não deve exigir a experiência ou o risco que exige atualmente. O Nginx se sai muito bem aqui, hospedando repositórios para as distribuições comuns de Linux para tornar a instalação mais fácil, e se a equipe do Apache (ou os fornecedores de distribuição Linux) não oferecerem algo semelhante, então eu só posso ver o uso do Apache continuando a diminuir enquanto luta para manter a relevância e mudar sua reputação de ser antigo e lento (com base em instalações mais antigas), embora as versões atualizadas tenham uma das melhores implementações do HTTP/2. Eu vejo isso como um problema menor para o IIS, já que geralmente é o servidor web preferido no lado do Windows.
```

6Apache -> Apache

**Review by eduqg — CHANGES_REQUESTED** [review]
Well done @elaynelemos!! Just little updates and it's ok for me

File: src/content/pt/2019/http2.md (hunk 7 above)
Comment (elaynelemos) [reply src/content/pt/2019/http2.md]: Done!

File: src/content/pt/2019/http2.md (hunk 9 above)
Comment (elaynelemos) [reply src/content/pt/2019/http2.md]: Solved!

**Comment by bazzadp** [comment]
And it's live: https://almanac.httparchive.org/pt/2019/http2

Welcome to the team @elaynelemos : https://almanac.httparchive.org/pt/2019/contributors#elaynelemos

**Comment by elaynelemos** [comment]
> And it's live: https://almanac.httparchive.org/pt/2019/http2
> 
> Welcome to the team @elaynelemos : https://almanac.httparchive.org/pt/2019/contributors#elaynelemos

This is the best part! Thank yooouu!

**Comment by bazzadp** [comment]
https://twitter.com/tunetheweb/status/1314608412886536192?s=20

**Pull request merged**